### ISSUE TITLE
Publish plugin zips to maven repo

### ISSUE THREAD

**Issue by prudhvigodithi** [opensearch-project/performance-analyzer-rca#175]
### Description

With existing setup the plugin jars are only published to maven repo, but not the generated plugin zips.

### Related META issue

https://github.com/opensearch-project/opensearch-build/issues/1916

### Current Behavior

Existing all maven publications target the jar files to publish to maven, but not the generated distribution zips.

### Expected Behavior

Along with plugin jars, plugin zips should also be published to maven repo with right maven coordinates, so user can download these plugins using dependancy model.

### Steps To Reproduce

1. Run the gradle assemble task that outputs jar and zips.
2. Using build setup the jars are published to local staging maven repo.
3. Zip's are just added to distribution's folder during build runtime.
4. Finally once build is done, just the jar files are added to local maven repo and updates to the manifest file, but not the zips.

### Proposed solution
A custom gradle plugin solution that can create the publication task for the zips to get published to local staging maven repo with right maven coordinates.
Add/Configure the custom gradle plugin openseearch.pluginzip to the gradle project and test the generated plugin zip's to upload to local maven repo.
More details on how to use the Plugin:
https://github.com/opensearch-project/opensearch-plugins/blob/main/BUILDING.md#opensearchpluginzip

**Comment by prudhvigodithi** [comment]
Hey @kkhatua @sgup432, just following up to see if `opensearch.pluginzip` is integrated, if not please make the necessary changes to add `opensearch.pluginzip` to support publishing your zips to Maven.

**Comment by sgup432** [comment]
@prudhvigodithi Not integrated yet. I will plan to raise the PR by today/tomorrow.

**Comment by sgup432** [comment]
@prudhvigodithi @bbarani I don't think we need to do this for performance-analyzer-rca package as this is not a plugin and runs as a separate process within cluster.
Main performance-analyzer plugin also take this as a dependency, so we should be covered as part of those changes.

**Comment by prudhvigodithi** [comment]
Hey @sgup432, make sense, if the component distribution output is not a zip, then `openseearch.pluginzip` is not required.

**Comment by prudhvigodithi** [comment]
Hey @sgup432 @kkhatua, just realized that if performance-analyzer plugin also take this(performance-analyzer-rca) as a dependency, then it should be part of maven, as the dependency will be fetched using maven coordinates.
@bbarani

**Comment by CEHENKLE** [comment]
@kkhatua Hey, what's the latest on this?  Do you have an ETA?

**Comment by sgup432** [comment]
@prudhvigodithi @CEHENKLE 
This RCA package(not a plugin) just act as a dependency and eventually bundled into main PA plugin. PA fetches RCA via git clone, unpacks and bundles it here - https://github.com/opensearch-project/performance-analyzer/blob/main/build.gradle#L432.

So these changes are not needed for this. Closing.
